purchase_discount/models/account_invoice_line.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    def new(self, values={}):
```

Comment:
Put here `values=None`, and check later:
```
values={} if values is None else values
```
Odoo is not doing a best practice here and can have side effects.